modules/core/shared/src/main/scala/lucuma/core/model/SpectralDistribution.scala

```diff
@@ -13,40 +13,135 @@ import eu.timepit.refined.cats._
 import eu.timepit.refined.types.numeric.PosBigDecimal
 import lucuma.core.enum.NonStellarLibrarySpectrum
 import lucuma.core.enum.StellarLibrarySpectrum
+import lucuma.core.math.BrightnessUnits._
+import lucuma.core.math.dimensional.GroupedUnitQty
+import monocle.Focus
+import monocle.Lens
+import monocle.Optional
+import monocle.Prism
+import monocle.macros.GenPrism
+import monocle.std.either._
 
-sealed trait SpectralDistribution extends Serializable
+// T = Brightness type (integral or surface); used only in EmissionLine
+sealed trait SpectralDistribution[+T] extends Serializable
 
 object SpectralDistribution {
 
-  /** A black body with a temperature in Kelvin. */
-  final case class BlackBody(temperature: Quantity[PosBigDecimal, Kelvin])
-      extends SpectralDistribution
+  /** A library defined spectrum. */
+  final case class Library(
+    // TODO The mockup at https://media.app.shortcut.com/api/attachments/files/clubhouse-assets/5f493112-dac1-417b-83f7-b20a9b5a40b6/61a51fbc-67bb-423c-b25b-f34640956c52/Explore_Targets_Source_Definition.png
+    // It doesn't mention non-stellar library. Is it another way of modeling Galaxy, Planet, Quasar, HII Region and Planetary Nebula?
```

Comment:
When do we clarify this?

Reply:
Per our direct conversation, we should remodel this. I'll make this the subject of the next PR.